examples/rcbilling-demo/src/pages/paywall/index.tsx

```diff
@@ -77,9 +113,17 @@ const PaywallPage: React.FC = () => {
       return;
     }
 
+    const offerId = pkg.rcBillingProduct.defaultSubscriptionPurchaseOptionId;
+    console.log(`Purchasing with option ${offerId}`);
+
     // How do we complete the purchase?
     try {
-      const { customerInfo } = await purchases.purchasePackage(pkg);
+      const { customerInfo } = offerId
+        ? await purchases.purchase({
+            rcPackage: pkg,
+            subscriptionPurchaseOptionId: offerId,
+          })
+        : await purchases.purchasePackage(pkg); // Testing backwards compatibility.
```

Comment:
Hmm I would probably avoid calling deprecated call here, since some people might be looking at this as sample code. I think the `purchasePackage` method should just call the `purchase` method directly, and we can have some unit tests to make sure it's doing the correct calls so I think it's ok not to test it here.